x-pack/plugins/security_solution/public/detection_engine/rule_creation/logic/esql_validator.ts

```diff
@@ -9,6 +9,8 @@ import { isEmpty } from 'lodash';
 import type { QueryClient } from '@tanstack/react-query';
 import { computeIsESQLQueryAggregating } from '@kbn/securitysolution-utils';
 
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
+import { validateQuery } from '@kbn/esql-validation-autocomplete';
 import { KibanaServices } from '../../../common/lib/kibana';
 
```

Comment:
there are multiple places when we need to use AST parser, to address https://github.com/elastic/security-team/issues/9282:

https://github.com/elastic/kibana/blob/8.15/x-pack/plugins/security_solution/public/detection_engine/rule_creation/logic/esql_validator.ts#L41

https://github.com/elastic/kibana/blob/8.15/packages/kbn-securitysolution-utils/src/esql/compute_if_esql_query_aggregating.ts

What is the intended scope of this PR?

Reply:
Good points!

Regarding the scope: we need to be able to a) get syntax errors if any while validating the query; b) switch to AST based utils to do query parsing.

Addressed points mentioned above in the new commit.

Reply:
done